pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -44,6 +44,14 @@
 # C can import package A propagated by B
 , propagatedBuildInputs ? []
 
+# Python module dependencies.
+# These are named after PEP-621.
+, dependencies ? []
+, optional-dependencies ? {}
+
+# Python PEP-517 build systems.
+, build-system ? []  # Note the naming being singular. This is in line with PEP-518.
```

Comment:
The comment dances around @phaer's point in above remarks that `build-system.requires` is the true list that we're modeling.